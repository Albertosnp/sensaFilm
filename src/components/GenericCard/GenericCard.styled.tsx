import styled from '@emotion/styled';
import breakpoints from '@/assets/styles/breakpoints';
import { rem } from '@/assets/styles/_mixins';

const GenericCardStyled = styled.figure`
  border-radius: 25px;
  height: auto;
  background-color: #e6e6e6;
  box-shadow: rgba(0, 0, 0, 0.15) 1.95px 1.95px 2.6px;
  transition: transform 0.5s;
  margin: 0 auto;
  /* height: 450px; */
  align-self: center;

  :hover {
    box-shadow: rgba(0, 0, 0, 0.25) 0px 54px 55px,
      rgba(0, 0, 0, 0.12) 0px -12px 30px, rgba(0, 0, 0, 0.12) 0px 4px 6px,
      rgba(0, 0, 0, 0.17) 0px 12px 13px, rgba(0, 0, 0, 0.09) 0px -3px 5px;

    transform: scale(1.1);
    z-index: 1;
  }

  .GenericCard {
    &__link {
      height: 100%;
      width: 100%;
      display: flex;
      flex-direction: column;
    }

    &__image {
      height: 100%;
      width: 100%;
      object-fit: cover;
      border-radius: 25px 25px 0 0;
      flex: 1 1 auto;
      max-height: 400px;
    }

    &__footer {
      display: flex;
      justify-content: space-between;
      padding: 15px;
      font-weight: 600;

      &__title {
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
        max-width: 200px;

        font-size: ${rem(14)};
        color: #232228;
      }
      &__votes {
        font-size: ${rem(16)};
      }
    }
  }
  @media (min-width: ${rem(breakpoints.tablet)}) {
  }
  @media (min-width: ${rem(breakpoints.tablet_wide)}) {
  }
  @media (min-width: ${rem(breakpoints.desktop)}) {
  }
`;
export default GenericCardStyled;
