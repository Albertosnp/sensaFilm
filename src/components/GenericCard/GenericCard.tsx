import { memo } from 'react';
import { Link } from 'react-router-dom';
import GenericCardStyled from './GenericCard.styled';

interface GenericCardProps {
  image: string;
  title: string;
  vote_average: number;
  path: string;
  id: number;
}
const GenericCard = ({
  image,
  title,
  vote_average,
  id,
  path,
}: GenericCardProps) => {
  return (
    <GenericCardStyled className="GenericCard">
      <Link to={`${path}/${id}`} className="GenericCard__link">
        <img
          className="GenericCard__image"
          src={`https://image.tmdb.org/t/p/w500/${image}`}
          alt="imagen-picture"
        />
        <div className="GenericCard__footer">
          <p className="GenericCard__footer__title">{title}</p>

          <span className="GenericCard__footer__votes">{vote_average}/10</span>
        </div>
      </Link>
    </GenericCardStyled>
  );
};

export default memo(GenericCard);
