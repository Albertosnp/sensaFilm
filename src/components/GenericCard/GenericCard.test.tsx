import React from 'react';
import { createRoot } from 'react-dom/client'
import { render } from '@testing-library/react';
import { vi } from 'vitest';
import GenericCard from './GenericCard';

vi.mock('react-router-dom', async () => {
  return {
    ...vi.importMock('react-router-dom'),
    useHistory: vi.fn(),
    useParams: vi.fn(),
    useLocation: () => ({
      search: '',
      pathname: '/',
    }),
    matchPath: vi.fn(),
    withRouter: vi.fn(),
    useRouteMatch: vi.fn(),
    Link: ({ children, to }: { children: JSX.Element; to: string }) =>
      React.createElement('a', { href: to }, children),
    Router: () => vi.fn(),
    HashRouter: () => vi.fn(),
    Switch: () => vi.fn(),
  };
});

describe('GenericCard', () => { 
  const defaultProps = {
    image: 'https://image.tmdb.org/t/p/w500',
    title: 'title',
    vote_average: 99,
    path: 'path',
    id: 1,
  };
  const getComponent = (props = defaultProps) => <GenericCard {...props} />;
  const setup = (props = defaultProps) => render(getComponent(props));

  it('Should render without crashing', () => {
    const div = document.createElement('div');
    createRoot(div).render(getComponent());
    createRoot(div).unmount();
  });

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
 })