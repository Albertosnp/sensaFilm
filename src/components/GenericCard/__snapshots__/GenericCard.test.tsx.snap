// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GenericCard > Should match snapshot 1`] = `
<div>
  <figure
    class="GenericCard css-1sbhet6"
  >
    <a
      href="path/1"
    >
      <img
        alt="imagen-picture"
        class="GenericCard__image"
        src="https://image.tmdb.org/t/p/w500/https://image.tmdb.org/t/p/w500"
      />
      <div
        class="GenericCard__footer"
      >
        <p
          class="GenericCard__footer__title"
        >
          title
        </p>
        <span
          class="GenericCard__footer__votes"
        >
          99
          /10
        </span>
      </div>
    </a>
  </figure>
</div>
`;
